mantatail.py

```diff
@@ -110,25 +110,55 @@ def recv_loop(state: ServerState, user_host: str, user_socket: socket.socket) ->
                         with state.lock:
                             call_handler_function(state, user, message)
 
-                    if user.closed_connection:
-                        return
-
 
 class UserConnection:
-    def __init__(self, host: str, socket: socket.socket, user_message: str, nick: str):
+    def __init__(self, state: ServerState, host: str, socket: socket.socket, user_message: str, nick: str):
+        self.state = state
+        self.send_que: queue.Queue[Tuple[str, str] | Tuple[None, None]] = queue.Queue()
         self.socket = socket
         self.host = host
         # Nick is shown in user lists etc, user_name is not
         self.nick = nick
         self.user_message = user_message
         self.user_name = user_message.split(" ", 1)[0]
         self.user_mask = f"{self.nick}!{self.user_name}@{self.host}"
-        self.closed_connection = False
+        self.que_thread = threading.Thread(target=self.start_queue_listener)
+        self.que_thread.start()
+
+    def start_queue_listener(self) -> None:
+        while True:
+            (message, prefix) = self.send_que.get()
 
-    def send_string_to_client(self, message: str, prefix: str = "mantatail") -> None:
+            if message is None or prefix is None:
+                self.send_quit_message()
+                self.state.delete_user(self.nick)
+                self.socket.close()
+                return print(f"{self.nick} has disconnected.")
```

Comment:
Because print always returns `None`, `return print("blah")` is correct, but it's confusing. Just write it on separate lines:
```suggestion
                print(f"{self.nick} has disconnected.")
                return
```